extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/tunnel/RestrictedGuacamoleTunnelService.java

```diff
@@ -180,8 +187,8 @@ protected ModeledConnection acquire(RemoteAuthenticatedUser user,
             @Override
             public int compare(ModeledConnection a, ModeledConnection b) {
 
-                return getActiveConnections(a).size()
-                     - getActiveConnections(b).size();
+                return ((getActiveConnections(a).size() + 1) * b.getConnectionWeight() -
```

Comment:
Repeating from an older comment for convenience: though that `+ 1` does result in the connection weight having an effect when the number of connections is zero, it also offsets the math such that the proportions governing weights in WLC are wrong, particularly for smaller numbers of connections.